packages/picasso/src/QuillEditor/formats/link.ts

```diff
@@ -0,0 +1,34 @@
+import { Classes } from '@toptal/picasso-shared'
+import Quill from 'quill'
+
+import getTypographyClassName from '../../Typography/utils/get-typography-class-name'
+
+const QuillLink = Quill.import('formats/link')
+
+class LinkBlot extends QuillLink {
+  static typographyClasses: Classes
+  static create(value: string) {
+    const node = super.create(value)
+
+    node.setAttribute(
+      'class',
+      getTypographyClassName(this.typographyClasses, {
+        variant: 'body',
+        size: 'inherit',
+        underline: 'solid',
+        color: 'light-blue',
+        weight: 'regular',
+      })
+    )
+
+    return node
+  }
+}
+
+const makeLinkFormat = (typographyClasses: Classes) => {
+  LinkBlot.typographyClasses = typographyClasses
+
+  return LinkBlot
+}
```

Comment:
This looks like a bug, we are setting a global variable with the argument and returning the global class. If this function is called in different places with different arguments, the second calling would override the first one.

If it's *really* necessary to generate a `class` with these dynamic fields, we could create it on the function calling. Just make sure to memorize it when been used, on React a simple `useMemo` should suffice

```suggestion
const makeLinkFormat = (typographyClasses: Classes) => class LinkBlot extends QuillLink {
  static create(value: string) {
    const node = super.create(value)

    node.setAttribute(
      'class',
      getTypographyClassName(typographyClasses, {
        variant: 'body',
        size: 'inherit',
        underline: 'solid',
        color: 'light-blue',
        weight: 'regular',
      })
    )

    return node
  }
}
```

Reply:
Augusto's suggestion makes sense to me 👍 and its more readable imho

Reply:
Sounds reasonable 👍

Reply:
https://toptal-core.atlassian.net/browse/FX-2893

Reply:
I will be marking the issue on Jira as a bug and changing its priority to one a bit higher